django/core/handlers/asgi.py

```diff
@@ -190,16 +190,28 @@ async def handle(self, scope, receive, send):
 
         async def process_request(request, send):
             response = await self.run_get_response(request)
-            await self.send_response(response, send)
+
+            try:
+                await self.send_response(response, send)
+            except asyncio.CancelledError:
+                # Client disconnected during send_response (ignore)
+                pass
+
+            return response
+
+        # Process request in async task
+        get_response_task = asyncio.create_task(process_request(request, send))
 
         # Try to catch a disconnect while getting response.
+        disconnect_task = asyncio.create_task(self.listen_for_disconnect(receive))
```

Comment:
Following the goal of having a minimal (yet correct) patch, would you consider moving this back to the `tasks` list, avoiding the definition of `disconnect_task`?

Also, I'm a bit concerned that changing the order of the tasks in `tasks` does not make any test fail, though that's unrelated to this patch. But if you have any idea on how to test that, we could definitely do a separated PR for it!